### ISSUE TITLE
Slow start of Alluxio client in Word Count

### ISSUE THREAD

**Issue by liupan664021** [Alluxio/alluxio#13118]
**Alluxio Version:**
2.4.1

**Describe the bug**
When I test wordcount task using mapreduce on alluxio, I found that it takes longer for map containers which read from alluxio to finish compared with those read from hdfs. More precisely, after the container starts, containers read from alluxio need about more 4~8 seconds to initialize and then read the first record.

It seems a slow start of alluxio client. Then I add some debug logs to figure out which steps alluxio delays. The logs tell that alluxio client takes about 4 seconds to load configurations with meta master.

Since the total cost of each map task is only about 40s, alluxio shortens the time to read data, but spends more time on initialization. So it seems that alluxio can not help with such mapreduce tasks.

Here is the debug code I add:
```java
# alluxio.util.ConfigurationUtils.java
public static GetConfigurationPResponse loadConfiguration(InetSocketAddress address,
      AlluxioConfiguration conf, boolean ignoreClusterConf, boolean ignorePathConf)
      throws AlluxioStatusException {
    GrpcChannel channel = null;
    try {
      LOG.debug("Alluxio client (version {}) is trying to load configuration from meta master {}",
          RuntimeConstants.VERSION, address);

      long start = System.currentTimeMillis();
      channel = GrpcChannelBuilder.newBuilder(GrpcServerAddress.create(address), conf)
          .setClientType("ConfigurationUtils").disableAuthentication().build();
      long stub = System.currentTimeMillis();
      LOG.info("########### build channel cost: " + (stub - start) + " ms");

      start = System.currentTimeMillis();
      MetaMasterConfigurationServiceGrpc.MetaMasterConfigurationServiceBlockingStub client =
          MetaMasterConfigurationServiceGrpc.newBlockingStub(channel);
      stub = System.currentTimeMillis();
      LOG.info("########### acquire client cost: " + (stub - start) + " ms");

      start = System.currentTimeMillis();
      GetConfigurationPResponse response = client.getConfiguration(
          GetConfigurationPOptions.newBuilder().setRawValue(true)
              .setIgnoreClusterConf(ignoreClusterConf).setIgnorePathConf(ignorePathConf).build());
      stub = System.currentTimeMillis();
      LOG.info("########### get configuration cost: " + (stub - start) + " ms");

      LOG.debug("Alluxio client has loaded configuration from meta master {}", address);

      return response;
    } catch (io.grpc.StatusRuntimeException e) {
      throw new UnavailableException(String.format(
          "Failed to handshake with master %s to load cluster default configuration values: %s",
          address, e.getMessage()), e);
    } catch (UnauthenticatedException e) {
      throw new RuntimeException(String.format(
          "Received authentication exception during boot-strap connect with host:%s", address),
          e);
    } finally {
      if (channel != null) {
        long start = System.currentTimeMillis();
        channel.shutdown();
        LOG.info("########### shut down channel cost: " + (System.currentTimeMillis() - start) + " ms");
      }
    }
  }
```
Here is the log of one map container:
...
2021-03-15 16:31:37,952 INFO [main] alluxio.util.ConfigurationUtils: ########### build channel cost: 220 ms
2021-03-15 16:31:37,958 INFO [main] alluxio.util.ConfigurationUtils: ########### acquire client cost: 6 ms
2021-03-15 16:31:39,666 INFO [main] alluxio.util.ConfigurationUtils: ########### get configuration cost: 1707 ms
2021-03-15 16:31:41,731 INFO [main] alluxio.util.ConfigurationUtils: ########### shut down channel cost: 2065 ms
...

Here are the yarn configuration:
- nodemanager: 19 nodes with 256GB memory and 96 processors
- maximum 96 containers per node
- no other load except test wordcount task
- word count size: 210GB
- split size is 128MB

Is there anyone who meet this problem before and can give me some suggestions? Thanks.

**To Reproduce**
Steps to reproduce the behavior (as minimally and precisely as possible)

**Expected behavior**
Figure out the problem and fix it.

**Urgency**
HIGH

**Additional context**

**Comment by ZacBlanco** [comment]
@liupan664021 I think all the costs up until the shutdown are necessary. It's possible they could be optimized, but setting up the channel and retrieving the configuration is necessary for the client in order for all cluster default configuration to be used. Currently this cannot be configured.

Shutting down the channel might arguably be something that can be done off the critical path in order to save those 2000ms or so. If you don't care for cluster defaults, then we could potentially avoid this whole process by adding a configuration flag to skip loading the remote configuration if the performance impact on the tasks is too large. Adding the configuration flag to skip loading cluster default configuration could essentially remove all of this overhead.

Unfortunately, the the client is quite heavyweight and is expected to be long running so that these costs don't add up. It sounds like with your current compute setup clients get created and destroyed frequently which makes the performance seem much worse. I believe there are other frameworks which you could use which don't create and destroy clients so frequently.

**Comment by apc999** [comment]
we discussed this issue offline and I believe the handshaking process between client and master to sync cluster config contributes to the slow start. @liupan664021  did you try to disable the flag "alluxio.user.conf.cluster.default.enabled" ?

I actually checked the source code and it seems this flag only controls conf sync between workers and master. I think this is a regression. But let us know your finding and it will be helpful
